template.yaml

```diff
@@ -110,4 +139,85 @@ Resources:
\ No newline at end of file
+      SourceArn: "arn:aws:execute-api:eu-west-2:761723964695:aqyf5d32s2/*/GET/.well-known/openid-configuration"
+
+  OpenIdConfigurationFunctionLogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub /aws/lambda/${Environment}-openid-configuration-lambda
+      KmsKeyId: !GetAtt MainKmsKey.Arn
+      RetentionInDays: 14
+
+  OpenIdConfigurationFunctionErrorMetricFilter:
+    Type: AWS::Logs::MetricFilter
+    Properties:
+      FilterName: !Sub ${Environment}-openid-configuration-errors
+      FilterPattern: "{($.level = \"ERROR\")}"
+      LogGroupName: !Ref OpenIdConfigurationFunctionLogGroup
+      MetricTransformations:
+        - MetricName: !Sub ${Environment}-openid-configuration-error-count
+          MetricNamespace: LambdaErrorsNamespace
+          MetricValue: 1
+
+  OpenIdConfigurationFunctionErrorCloudwatchAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      ActionsEnabled: true
+      AlarmActions:
+        - !Ref OpenIdConfigurationFunctionSlackEvents
+      AlarmDescription: !Sub "10 or more errors have occurred in the ${Environment} openid-configuration lambda.ACCOUNT: ${AWS::AccountId}"
+      AlarmName: !Sub ${Environment}-openid-configuration-alarm
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      EvaluationPeriods: 1
+      MetricName: OpenIdConfigurationFunctionErrorMetricFilter.MetricTransformations[0].MetricName
+      Namespace: OpenIdConfigurationFunctionErrorMetricFilter.MetricTransformations[0].MetricNamespace
+      Period: 3600
+      Statistic: Sum
+      Threshold: 10
+
+
+  OpenIdConfigurationFunctionErrorRateCloudwatchAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmName: !Sub ${Environment}-openid-configuration-error-rate-alarm
+      AlarmDescription: !Sub "Lambda error rate of 10 has been reached in the ${Environment} openid-configuration lambda.ACCOUNT: ${AWS::AccountId}"
+      ActionsEnabled: true
+      AlarmActions:
+        - !Ref OpenIdConfigurationFunctionSlackEvents
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      EvaluationPeriods: 1
+      Threshold: 10
+      Metrics:
+        - Id: e1
+          Expression: m2/m1*100
+          Label: Error Rate
+          ReturnData: true
+        - Id: m1
+          MetricStat:
+            Metric:
+              Namespace: AWS/Lambda
+              MetricName: Invocations
+              Dimensions:
+                - Name: FunctionName
+                  Value: !Ref OpenIdConfigurationFunction.Alias
+            Period: 60
+            Stat: Sum
+            Unit: Count
+        - Id: m2
+          MetricStat:
+            Metric:
+              Namespace: AWS/Lambda
+              MetricName: Errors
+              Dimensions:
+                - Name: FunctionName
+                  Value: !Ref OpenIdConfigurationFunction.Alias
+            Period: 60
+            Stat: Sum
+            Unit: Count
+
+
+  OpenIdConfigurationFunctionSlackEvents:
```

Comment:
I think this resource name is wrong. We're only going to have one of these, not one per lambda right?

Reply:
Ah yes, good point, shall change that now